### DISCUSSION TITLE
TTL implementation

### DISCUSSION THREAD

**Discussion by byte-sourcerer** [ClickHouse/ClickHouse#66078] [Q&A] [answered]
I'm interested in delving deeper into the intricacies of the TTL mechanism's implementation. Are there any resources or code references that I could possibly review?

**Comment by yiyang-shao** [comment, marked as answer]
Hi!

Do you have more specific question? If not, you can take the [documentation](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#ttl) as a starting point.

In addition, to see how TTL is implemented in the ClickHouse source code. you may specifically, look at the MergeTree family of table engines, as TTL is a feature of these engines.
Key files to review:
	•	`MergeTreeDataPart.cpp` and `MergeTreeDataPart.h`: These files handle data parts, which include the logic for TTL.
	•	`TTLDescription.cpp` and `TTLDescription.h`: These files describe the TTL settings.
	•	`MergeTreeData.cpp` and `MergeTreeData.h`: These files include the logic for managing data in the MergeTree family tables.

Thank you!
